2-0-data-structures-and-algorithms/src/test/java/com/bobocode/stack/StackTest.java

```diff
@@ -1,88 +1,355 @@
+    }
+
+    @Test
+    @Order(19)
     void isEmpty() {
-        intStack = LinkedStack.of(87, 53, 66);
+        fillTestStack(87, 53, 66);
 
         boolean stackEmpty = intStack.isEmpty();
 
         assertThat(stackEmpty).isEqualTo(false);
     }
 
     @Test
-    @Order(8)
+    @Order(20)
     void isEmptyOnEmptyStack() {
         boolean stackEmpty = intStack.isEmpty();
-
         assertThat(stackEmpty).isEqualTo(true);
     }
+
+    private Class<?> getInnerClass() {
+        return Arrays.stream(LinkedStack.class.getDeclaredClasses())
+                .filter(Class::isMemberClass)
+                .findAny().orElseThrow();
+    }
+
+    private Field getHeadField() {
+        Field headField = Arrays.stream(LinkedStack.class.getDeclaredFields())
+                .filter(NODE_FIELD_PREDICATE)
+                .findAny()
+                .orElseThrow();
+        headField.setAccessible(true);
+        return headField;
+    }
+
+    private Field getNodeElementField(Object node) {
+        Field fieldElement = Arrays.stream(node.getClass().getDeclaredFields())
+                .filter(NODE_ELEMENT_FIELD)
+                .findAny()
+                .orElseThrow();
+        fieldElement.setAccessible(true);
+        return fieldElement;
+    }
+
+    private Field getNodeNextField(Object node) {
+        Field field = Arrays.stream(node.getClass().getDeclaredFields())
+                .filter(NODE_NEXT_FIELD)
+                .findAny()
+                .orElseThrow();
+        field.setAccessible(true);
+        return field;
+    }
+
+    @SneakyThrows
+    private Object getHeadObject() {
+        return getHeadField().get(intStack);
+    }
+
+    @SneakyThrows
+    private int getNodeElementInt(Object node) {
+        return (int) getNodeElementField(node).get(node);
+    }
+
+    @SneakyThrows
+    private Object getNodeNextObject(Object node) {
+        return getNodeNextField(node).get(node);
+    }
+
+    private boolean contains(int element) {
+        Object head = getHeadObject();
+        if (head == null) {
+            return false;
+        }
+
+        if (getNodeNextObject(head) != null) {
+            return checkNext(head, element);
+        } else {
+            return getNodeElementInt(head) == element;
+        }
+    }
+
+    private boolean checkNext(Object node, int element) {
+        if (getNodeElementInt(node) == element) {
+            return true;
+        } else {
+            return checkNext(getNodeNextObject(node), element);
+        }
+    }
+
+    @SneakyThrows
+    private Object newNode(int element) {
+        Constructor<?> constructor = getInnerClass().getDeclaredConstructors()[0];
+        constructor.setAccessible(true);
+
+        if (constructor.getParameters().length == 1) {
+            return constructor.newInstance(element);
+        } else {
+            Object node = constructor.newInstance();
+            getNodeElementField(node).set(intStack, element);
```

Comment:
In this case your implementation can set only into `Stack` objects. Pay attention for the object you pass into `set()` parameter.